### ISSUE TITLE
Questions about HSN tutorial

### ISSUE THREAD

**Issue by georg-bn** [pyt-team/TopoModelX#130]
I have a couple of questions/found bugs regarding the HSN tutorial (and hence might impact other tutorials in the simplicial domain).

1. https://github.com/pyt-team/TopoModelX/blob/18956deb499c58062b0d435df8ddc85fc13b6634/tutorials/simplicial/hsn_train.ipynb#L326 should be `self.layers = torch.nn.ModuleList(layers)`, so that the parameters get properly registered.
2. https://github.com/pyt-team/TopoModelX/blob/18956deb499c58062b0d435df8ddc85fc13b6634/tutorials/simplicial/hsn_train.ipynb#L355 should probably not have `softmax`, as later binary crossentropy on logits is used: https://github.com/pyt-team/TopoModelX/blob/18956deb499c58062b0d435df8ddc85fc13b6634/tutorials/simplicial/hsn_train.ipynb#L415
3. https://github.com/pyt-team/TopoModelX/blob/18956deb499c58062b0d435df8ddc85fc13b6634/tutorials/simplicial/hsn_train.ipynb#L120 *"Here, we have in_channels = channels_nodes $ = 34$. This is because the Karate dataset encodes the identity of each of the 34 nodes as a one hot encoder."* This seems to be incorrect as we get 2 dim features: https://github.com/pyt-team/TopoModelX/blob/18956deb499c58062b0d435df8ddc85fc13b6634/tutorials/simplicial/hsn_train.ipynb#L145 and they are eigenvectors from the graph as defined in https://github.com/pyt-team/TopoNetX/blob/4c47ec24047a7af83d5a249a79c1945e7043ceea/toponetx/datasets/graph.py#L38 .
